libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKUpgradeManager.java

```diff
@@ -96,17 +84,16 @@ protected synchronized void upgradeAccMgr() {
         if (TextUtils.isEmpty(installedVersion)) {
             installedVersion = getInstalledAccMgrVersion();
         }
-
-        /*
-         * If the installed version < v2.2.0, we need to store the current
-         * user's user ID and org ID in a shared preference file, to
-         * support fast user switching.
-         */
         try {
-        	final String majorVersionNum = installedVersion.substring(0, 3);
+            final String majorVersionNum = installedVersion.substring(0, 3);
             double installedVerDouble = Double.parseDouble(majorVersionNum);
-            if (installedVerDouble < 2.2) {
-            	upgradeTo2Dot2();
```

Comment:
I don't think so. If they upgrade from pre-2.2 directly to 6.0, there's no recovery path anyway.